### PR TITLE
CUDA-Q quantum transformer tutorial

### PR DESCRIPTION
**Pull request by anthonysmaldone**
### Description

This is a `ipynb` notebook and corresponding scripts that include a tutorial of a hybrid quantum-classical transformer model from the paper: A Hybrid Transformer Architecture with a Quantized Self-Attention Mechanism Applied to Molecular Generation - a collaborative effort between NVIDIA and Yale University.

### CODE REVIEW

**Comment by anthonysmaldone** [comment]
@sacpis I have fixed the installation issue and restyled the code. There are basic machine learning and chemistry terms that fail during spellcheck (such as 'logits', 'dataset', 'stereocenters', 'physicochemical', etc) should I add some of these to the spelling_allowlist.txt in the workflow config?

**Comment by anthonysmaldone** [comment]
@sacpis Yes, they should all be in comments/docstrings

**Comment by anthonysmaldone** [comment]
@sacpis I have made the requested changes, please let me know if there is anything else I can do.

**Comment by anthonysmaldone** [comment]
@sacpis I've added the notebook to the application.rst file

**Comment by bmhowe23** [comment]
> @bmhowe23 Is it safe to skip the conversion of quantum_transformer notebook? It is taking long and failing. https://github.com/NVIDIA/cuda-quantum/actions/runs/13591336814/job/37999481528?pr=2614

That should probably be answered by @bettinaheim. We normally like to be able to test our published examples on a regular basis to ensure they don't break with any new cuda-quantum code updates. Skipping testing of this notebook would expose us to the risk of it breaking and us not knowing about it.

Are you proposing skipping testing of this in just the per-PR CI, all CI, or all automated testing (even release-level QA testing)?

**Comment by anthonysmaldone** [comment]
@sacpis After checking the docs preview, I noticed the sections of the transformer notebook are headings under "Applications", let if know if I should change the notebook

**Comment by anthonysmaldone** [comment]
@sacpis It should still be under 'Applications". However now I see that Applications has subheadings "Installation", "Algorithm and Examples" and "Usage" -which are the section titles in the notebook. Do they need to be one level smaller in the tutorial notebook?

**Comment by anthonysmaldone** [comment]
@sacpis Sounds good, ive made the updates

**Comment by bmhowe23** [comment]
> Trying to optimize the training process as I have only 1 GPU locally and CI runs on CPU.

I would recommend putting `cudaq.set_target('nvidia')` in the notebook. The script that runs the CI (docs/notebook_validation.py) will notice that and skip it if the target is not present on the host that it is running on.
